pkg/trace/api/otlp.go

```diff
@@ -302,10 +304,23 @@ func (o *OTLPReceiver) receiveResourceSpansV2(ctx context.Context, rspans ptrace
 		hostname = o.conf.Hostname
 		src = source.Source{Kind: source.HostnameKind, Identifier: hostname}
 	}
-	containerID := traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeContainerID, semconv.AttributeK8SPodUID)
+	if o.conf.OTLPReceiver.IgnoreMissingDatadogFields {
+		hostname = ""
+	}
+	if incomingHostname := traceutil.GetOTelAttrVal(otelres.Attributes(), true, "datadog.host"); incomingHostname != "" {
```

Comment:
Nit: might be worth having all of these `datadog.host`, `datadog.container_id`, etc... behind some kind of type alias for better code maintenance.